cart.py

```diff
@@ -359,6 +360,21 @@ def add(lang):
         abort(404)
     website, = websites
 
+    cursor = Transaction().connection.cursor()
+
+    if session.get('customer'):
+        lock_id = ('1%%0%sd' % 7 ) % int(str(session['customer'])[:7])
+    else:
+        lock_id = int('2'+str(uuid.UUID(session.sid).int)[:7])
```

Comment:
AFAIU uuid.UUID will create a very larg integer which may not be supported by PostgreSQL:

https://www.postgresql.org/docs/current/functions-admin.html#FUNCTIONS-ADVISORY-LOCKS
https://www.postgresql.org/docs/current/datatype-numeric.html

Reply:
I didn't realize you were already picking just the first 7 characters. So I think the code is ok as is.